src/pyate/term_extraction_pipeline.py

```diff
@@ -20,23 +22,26 @@ def __init__(
         self.args = args
         self.kwargs = kwargs
         self.__name__ = self.func.__name__
+        self.matcher = Matcher(nlp.vocab)
         Doc.set_extension(self.__name__, default=None, force=force)
-
-    def __call__(self, doc: Doc):
-        term_counter = defaultdict(int)
+        self.term_counter = None
 
         def add_to_counter(matcher, doc, i, matches) -> Doc:
             match_id, start, end = matches[i]
             candidate = str(doc[start:end])
             if TermExtraction.word_length(candidate) <= TermExtraction.MAX_WORD_LENGTH:
-                term_counter[candidate] += 1
+                self.term_counter[candidate] += 1
 
         for i, pattern in enumerate(TermExtraction.patterns):
-            TermExtraction.matcher.add("term{}".format(i), add_to_counter, pattern)
-        matches = TermExtraction.matcher(doc)
```

Comment:
I changed the way the Matcher was initialized because I noticed that the TermExtraction.matcher is always using the English model. But, if a user does something like:
```Python
nlp = spacy.load('de_core_news_sm')
nlp.add_pipe(TermExtractionPipeline(func))
```
then, the Matcher would use the English vocab (as defined in the TermExtraction class) and not the German Vocabulary. Hence, I added the `nlp` object as parameter to `TermExtractionPipeline`'s `__init__` and initialized the Matcher as `self.matcher = Matcher(nlp.vocab)`

Reply:
Awesome, thanks @stelmath. Sorry I didn't see this message earlier. I will merge the PR now.